### PR TITLE
Automated cherry pick of #3081: Only watch the monitor CR if the CRD exists.

### PR DESCRIPTION
**Pull request by rene-dekker**
Cherry pick of #3081 on release-v1.33.

#3081: Only watch the monitor CR if the CRD exists.

### CODE REVIEW

**Review by hjiawei — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
